packages/transactional-adapters/transactional-adapter-pg-promise/test/transactional-adapter-pg-promise.spec.ts

```diff
@@ -0,0 +1,211 @@
+            'SELECT * FROM public.user WHERE id = $1',
+            [id],
+        );
+    }
+
+    async createUser(name: string) {
+        const created = await this.txHost.tx.one<UserRecord>(
+            'INSERT INTO public.user (name, email) VALUES ($1, $2) RETURNING *',
+            [name, `${name}@email.com`],
+        );
+        return created;
+    }
+}
+
+@Injectable()
+class UserService {
+    constructor(
+        private readonly userRepository: UserRepository,
+        private readonly txHost: TransactionHost<TransactionalAdapterPgPromise>,
+        @Inject(PG_PROMISE)
+        private readonly db: Database,
+    ) {}
+
+    @Transactional()
+    async transactionWithDecorator() {
+        const r1 = await this.userRepository.createUser('John');
+        const r2 = await this.userRepository.getUserById(r1.id);
+        return { r1, r2 };
+    }
+
+    @Transactional<TransactionalAdapterPgPromise>({ mode: transactionMode })
+    async transactionWithDecoratorWithOptions() {
+        const r1 = await this.userRepository.createUser('James');
+        const r2 = await this.db.oneOrNone<UserRecord>(
+            'SELECT * FROM public.user WHERE id = $1',
+            [r1.id],
+        );
+        const r3 = await this.userRepository.getUserById(r1.id);
+        return { r1, r2, r3 };
+    }
+
+    async transactionWithFunctionWrapper() {
+        return this.txHost.withTransaction(
+            { mode: transactionMode },
+            async () => {
+                const r1 = await this.userRepository.createUser('Joe');
+                const r2 = await this.db.oneOrNone<UserRecord>(
+                    'SELECT * FROM public.user WHERE id = $1',
+                    [r1.id],
+                );
+                const r3 = await this.userRepository.getUserById(r1.id);
+                return { r1, r2, r3 };
+            },
+        );
+    }
+
+    @Transactional()
+    async transactionWithDecoratorError() {
+        await this.userRepository.createUser('Nobody');
+        throw new Error('Rollback');
+    }
+}
+
+@Module({
+    providers: [
+        {
+            provide: PG_PROMISE,
+            useValue: db,
+        },
+    ],
+    exports: [PG_PROMISE],
+})
+class PgPromiseModule {}
+
+@Module({
+    imports: [
+        PgPromiseModule,
+        ClsModule.forRoot({
+            plugins: [
+                new ClsPluginTransactional({
+                    imports: [PgPromiseModule],
+                    adapter: new TransactionalAdapterPgPromise({
+                        dbInstanceToken: PG_PROMISE,
+                    }),
+                }),
+            ],
+        }),
+    ],
+    providers: [UserService, UserRepository],
+})
+class AppModule {}
+
+describe('Transactional', () => {
+    let module: TestingModule;
+    let callingService: UserService;
+
+    beforeAll(async () => {
+        execSync(
+            'docker-compose -f test/docker-compose.yml up -d --quiet-pull --wait',
```

Comment:
tests are failing because of `docker-compose`. Use `docker compose instead` (without the hyphen)